### PR TITLE
Update docs for versioning for the website

### PR DESCRIPTION
**Pull request by TOsmanov**
### Summary

This PR contains updates documentation adapted for the website from PR #393

- Fix offset of header anchors
- Mergences the "models" section in one file
- `docs` and `images` folders are now deleted before `git checkout` because files that are not in the previous version remain after the site is generate

#### Preview:
https://tosmanov.github.io/datumaro/docs/

### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have added description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md)
- [x] I have updated the [documentation](
  https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly
- [ ] I have added tests to cover my changes
- [x] I have [linked related issues](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2021 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: site/content/en/docs/developer_manual.md [hunk 1]
```diff
@@ -127,9 +126,8 @@ reduce the amount of disk writes. Changes can be flushed with
 `flush_changes()`.
 
 ```python
-from datumaro.components.annotation import Bbox, Polygon
 from datumaro.components.dataset import Dataset
-from datumaro.components.extractor import DatasetItem
+from datumaro.components.extractor import Bbox, Polygon, DatasetItem
```
Comment (zhiltsov-max) [inline site/content/en/docs/developer_manual.md:130]: The previous variant is correct.

File: .github/workflows/github_pages.yml [hunk 2]
```diff
@@ -32,7 +32,7 @@ jobs:
 
       - name: Build docs
         run: |
-          pip install gitpython packaging toml
+          pip install gitpython packaging toml shutil
```
Comment (zhiltsov-max) [inline .github/workflows/github_pages.yml:35]: `shutil` is a standard library module, why is it needed here?

File: site/content/en/docs/getting_started.md [hunk 3]
```diff
@@ -26,9 +26,11 @@ python -m virtualenv venv
...
 ``` bash
-pip install datumaro
+pip install datumaro[default]
 ```
 
+Read full installation instructions in [the user manual](/docs/user-manual/installation).
+
 ## Usage
 
 There are several options available:
```
Comment (zhiltsov-max) [inline site/content/en/docs/getting_started.md:36]: Please, update the example below with:

``` python
from datumaro.components.project import Project

# load a Datumaro project
project = Project('directory')

# create a dataset
dataset = project.working_tree.make_dataset()

# keep only annotated images
dataset.select(lambda item: len(item.annotations) != 0)

# change dataset labels
dataset.transform('remap_labels',
  {'cat': 'dog', # rename cat to dog
    'truck': 'car', # rename truck to car
    'person': '', # remove this label
  }, default='delete') # remove everything else

# iterate over dataset elements
for item in dataset:
  print(item.id, item.annotations)

# export the resulting dataset in COCO format
dataset.export('dst/dir', 'coco')

# optionally, release the project resources
project.close()
```

**Comment by zhiltsov-max** [comment]
image

I feel, we should indicate somehow, that the "latest version" is the current development branch, not an actual release. The logic slightly differs from CVAT, because there users sit on the development branch, while we release packages on PyPI, so the latest version is the latest *released* version in our case.

File: site/content/en/docs/user-manual/command-reference/_index.md [hunk 4]
```diff
@@ -2,43 +2,76 @@
...
+The command line is split into the separate _commands_ and command _contexts_.
+Contexts group multiple commands related to a specific topic, e.g.
+project operations, data source operations etc. Almost all the commands
+operate on projects, so the `project` context and commands without a context
+are mostly the same. By default, commands look for a project in the current
+directory. If the project you're working on is located somewhere else, you
+can pass the `-p/--project <path>` argument to the command.
+
+> **Note**: command behavior is subject to change and might be outdated,
+> **always check the `--help` output of the specific command**
```
Comment (zhiltsov-max) [inline site/content/en/docs/user-manual/command-reference/_index.md:62]:
```suggestion
> **Note**: command behavior is subject to change, so this text might be outdated,
> **always check the `--help` output of the specific command**
```

File: site/content/en/docs/user-manual/command-reference/checkout.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+This command allows to restore a specific project revision in the project
+tree or to restore separate revisions of sources. A revision can be a commit
+hash, branch, tag, or any [relative reference in the Git format](https://git-scm.com/book/en/v2/Git-Tools-Revision-Selection).
+
+This command has multiple forms:
+``` bash
+1) datum checkout \<revision\>
+2) datum checkout [--] \<source1\> ...
+3) datum checkout \<revision\> [--] \<source1\> \<source2\> ...
```
Comment (zhiltsov-max) [inline site/content/en/docs/user-manual/command-reference/checkout.md:16]: image
```suggestion
1) datum checkout <revision>
2) datum checkout [--] <source1> ...
3) datum checkout <revision> [--] <source1> <source2> ...
```

**Comment by zhiltsov-max** [comment]
Please, update the contributing page:
 
1. 

Then install all dependencies:

``` bash
pip install -r requirements.txt
```

2. Remove this fragment:
image

File: site/content/en/docs/user-manual/command-reference/_index.md [hunk 6]
```diff
@@ -2,43 +2,77 @@
...
+  classDef nofillclass fill-opacity:0;
+  classDef hideclass fill-opacity:0,stroke-opacity:0;
+  classDef filloneclass fill:#CCCCFF,stroke-opacity:0;
+  classDef filltwoclass fill:#FFFF99,stroke-opacity:0;
+  classDef fillthreeclass fill:#CCFFFF,stroke-opacity:0;
+  classDef fillfourclass fill:#CCFFCC,stroke-opacity:0;
 ```
 
 </div>
+
```
Comment (zhiltsov-max) [inline site/content/en/docs/user-manual/command-reference/_index.md:52]: I can't see the image here:
image

No errors in the log.

**Comment by zhiltsov-max** [comment]
The "project" CLI context is missing on the diagram, please add "Project - info" branch there.

**Comment by TOsmanov** [comment]
> The "project" CLI context is missing on the diagram, please add "Project - info" branch there.

@zhiltsov-max  Just like that?

image

**Comment by zhiltsov-max** [comment]
Right

**Review by zhiltsov-max — APPROVED** [review]
Thank you for the work done!

**Pull request merged**